File: 0387.字符串中的第一个唯一字符.java
/*
 * @lc app=leetcode.cn id=387 lang=java
 *
 * [387] 字符串中的第一个唯一字符
 *
 * https://leetcode-cn.com/problems/first-unique-character-in-a-string/description/
 *
 * algorithms
 * Easy (41.54%)
 * Likes:    164
 * Dislikes: 0
 * Total Accepted:    53.6K
 * Total Submissions: 127.1K
 * Testcase Example:  '"leetcode"'
 *
 * 给定一个字符串，找到它的第一个不重复的字符，并返回它的索引。如果不存在，则返回 -1。
 * 
 * 案例:
 * 
 * 
 * s = "leetcode"
 * 返回 0.
 * 
 * s = "loveleetcode",
 * 返回 2.
 * 
 * 
 * 
 * 
 * 注意事项：您可以假定该字符串只包含小写字母。
 * 
 */

// @lc code=start
import java.util.HashMap;

class Solution {
    public int firstUniqChar(String s) {
        char[] arr = s.toCharArray();
        HashMap<Character, Integer> map = new HashMap<>();
        for (char c : arr) {
            if (!map.containsKey(c)) {
                map.put(c, 1);
            } else {
                map.put(c, map.get(c) + 1);
            }
        }

        int res = -1;
        for (int i = 0; i < arr.length; i++) {
            if (map.get(arr[i]) == 1) {
                res = i;
                break;
            }
        }
        return res;
    }
}
// @lc code=end

File: 0387.字符串中的第一个唯一字符.cs
/*
 * @lc app=leetcode.cn id=387 lang=csharp
 *
 * [387] 字符串中的第一个唯一字符
 *
 * https://leetcode-cn.com/problems/first-unique-character-in-a-string/description/
 *
 * algorithms
 * Easy (47.54%)
 * Likes:    332
 * Dislikes: 0
 * Total Accepted:    139K
 * Total Submissions: 276.1K
 * Testcase Example:  '"leetcode"'
 *
 * 给定一个字符串，找到它的第一个不重复的字符，并返回它的索引。如果不存在，则返回 -1。
 * 
 * 
 * 
 * 示例：
 * 
 * s = "leetcode"
 * 返回 0
 * 
 * s = "loveleetcode"
 * 返回 2
 * 
 * 
 * 
 * 
 * 提示：你可以假定该字符串只包含小写字母。
 * 
 */

// @lc code=start
using System.Collections;

public class Solution
{
    public int FirstUniqChar(string s)
    {
        Hashtable tmp = new Hashtable();
        foreach (char ch in s)
        {
            if (!tmp.Contains(ch))
                tmp.Add(ch, 1);
            else
                tmp[ch] = (int)tmp[ch] + 1;
        }
        for (int i = 0; i < s.Length; ++i)
        {
            if ((int)tmp[s[i]] == 1)
                return i;
        }
        return -1;
    }
}

// @lc code=end

